tests/hpc/slurm_master_db.pm

```diff
@@ -18,6 +18,165 @@ use warnings;
 use testapi;
 use lockapi;
 use utils;
+use Data::Dumper;
+use Mojo::JSON;
+
+## TODO: provide better parser for HPC specific tests
+sub get_test_name {
+    return my $name = (caller(2))[3];
+}
+
+sub validate_result {
+    my ($result) = @_;
+
+    if ($result == 0) {
+        return 'PASS';
+    } elsif ($result == 1) {
+        return 'FAIL';
+    } else {
+        return undef;
+    }
+}
+
+sub results {
+    my ($description, $result) = @_;
+
+    my %results = (
+        test        => get_test_name(),
+        description => $description,
+        result      => validate_result($result)
+    );
+    return %results;
+}
+
+sub pars_results {
+    my (%test)  = @_;
+    my $file    = 'tmpresults.xml';
+    my @results = keys %test;
+
+    my $test_name;
+    my $description;
+    my $result;
+
+    for my $i (@results) {
+        if ($i eq 'test') {
+            $test_name = $test{$i};
+        } elsif ($i eq 'description') {
+            $description = $test{$i};
+        } elsif ($i eq 'result') {
+            if ($test{$i} eq 'FAIL') {
+                $result = $test{$i};
+            }
+        } else {
+            die('Wrong key!');
+        }
+    }
+
+    if ($result eq 'FAIL') {
+        script_run("echo \"<testcase name='$test_name' errors='1'>\" | tee -a $file");
+    } else {
+        script_run("echo \"<testcase name='$test_name'>\" | tee -a $file");
+    }
+    script_run("echo \"<system-out>\" | tee -a $file");
+    script_run("echo $description | tee -a $file");
+    script_run("echo \"</system-out>\" | tee -a $file");
+    script_run("echo \"</testcase>\" | tee -a $file");
+}
+
+sub test_01 {
+    my $description = 'Basic SRUN test';
+    my $result;
+
+    $result = script_run("srun -w slave-node00 date");
+
+    my %results = results($description, $result);
+    return %results;
+}
+
+sub test_02 {
+    my $description = 'Simple SINFO test';
+    my $result;
+
+    $result = script_run('sinfo');
+
+    ##TODO: add check of sinfo vs slurm.conf?
+
+    my %results = results($description, $result);
+    return %results;
+}
+
+##TODO: improve before enabling
+sub test_03 {
+    my $description = 'SRUN stress test';
+    my $result      = 0;
+    my $result0;
+    my $result1;
+
+    my @collected_results;
+
+    for (my $i = 0; $i < 50; $i++) {
+        $result0 = script_run("srun -w slave-node00 date");
```

Comment:
because you not actually counting how many times each command is failing and if both or only one is failing I think this could be simplified to : 

```
for (my $i = 0; $i < 50; $i++) { 
return  results($description, 1)    if ( script_run("srun -w slave-node00 date") == 1 );
return  results($description, 1)    if ( script_run("srun -w slave-node01 date") == 1 );
}
```

and you can drop everything below and above in this test 
}

Reply:
yes, that would simplify a lot of code. 

I would go further and avoid comparing a result with 1. I don't know about `srun`, but sometimes some commands produce different error codes. I suggest you compare `!=0`

for (my $i = 0; $i < 50; $i++) { 
    return  results($description, 1)    if ( script_run("srun -w slave-node00 date") != 0 );
    return  results($description, 1)    if ( script_run("srun -w slave-node01 date") != 0 );
}

Reply:
as above. This was done in a few minutes. Overall I replace it with TODO now to have proper stress test for the cluster.